src/serverExplorer.ts

```diff
@@ -94,11 +95,32 @@ export class ServersViewTreeDataProvider implements TreeDataProvider<Protocol.Se
             canSelectMany: false,
             canSelectFolders: true,
             openLabel: 'Select desired server location'
-        }).then(folders => {
+        })
+        .then(folders => {
             if (folders && folders.length === 1) {
                 return this.client.findServerBeans(folders[0].fsPath);
             }
-        }).then(serverBeans => {
+        })
+        .then(this.getServerName())
+        .then((value) => this.getRequiredParameters(value))
+        .then((value) => this.getOptionalParameters(value))
+        .then(this.createServerAsync());
+    }
+
+    private createServerAsync(): (value: { name: string; bean: Protocol.ServerBean; attributes: {}}) => Thenable<Protocol.Status> {
+        return async (data) => {
+            if (data && data.name) {
+                const status = await this.client.createServerAsync(data.bean, data.name, data.attributes);
+                if (status.severity > 0) {
+                    return Promise.reject(status.message);
+                }
+                return Promise.resolve(status);
```

Comment:
Creating fulfilled issue is not required here and in several places below, because then will do that anyway for value returned from callback IIRC.
From [then doc](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/then):
> returns a value, the promise returned by then gets resolved with the returned value as its value